# curran phillips

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from mlxtend.preprocessing import minmax_scaling
import seaborn as sns
import matplotlib.pyplot as plt
from matplotlib.colors import ListedColormap
from sklearn.datasets import make_circles, make_classification, make_moons
from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
from sklearn.ensemble import AdaBoostClassifier, RandomForestClassifier
from sklearn.gaussian_process import GaussianProcessClassifier
from sklearn.gaussian_process.kernels import RBF
from sklearn.inspection import DecisionBoundaryDisplay
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.neighbors import KNeighborsClassifier
from sklearn.neural_network import MLPClassifier
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier
from sklearn import svm
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix

In [2]:
df = pd.read_csv('../../Combine_Data/men/pb_encoded.csv')
olymp = pd.read_csv('../../Data/cleandata22-23/encoded_m_olympics_pb.csv')
pbnames = pd.read_csv('../../Data/cleandata22-23/encoded_m_olympics_pbnames.csv')
pbolymp = pd.read_csv('../../Data/cleandata22-23/men22_23.csv')


In [3]:
usaa=pbolymp[pbolymp['Country']=='USA']

In [4]:
usaa.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1463 entries, 525 to 8557
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         1463 non-null   object 
 1   Country      1463 non-null   object 
 2   Date         1463 non-null   object 
 3   Gender       1463 non-null   object 
 4   Competition  1463 non-null   object 
 5   Round        1463 non-null   object 
 6   Location     1463 non-null   object 
 7   Apparatus    1463 non-null   object 
 8   Rank         1463 non-null   float64
 9   D_Score      1463 non-null   float64
 10  E_Score      1463 non-null   float64
 11  Penalty      1463 non-null   float64
 12  Score        1463 non-null   float64
 13  Total_Score  1463 non-null   float64
dtypes: float64(6), object(8)
memory usage: 171.4+ KB


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Rank         367 non-null    int64  
 1   D            367 non-null    float64
 2   E            367 non-null    float64
 3   ND           367 non-null    float64
 4   Total        367 non-null    float64
 5   year         367 non-null    int64  
 6   medal        367 non-null    int64  
 7   Name         367 non-null    int64  
 8   Nation       367 non-null    int64  
 9   round_final  367 non-null    bool   
 10  round_qual   367 non-null    bool   
dtypes: bool(2), float64(4), int64(5)
memory usage: 26.6 KB


In [6]:

# Assuming you already have your DataFrame 'df' and the target variable is 'medal'
y = df['medal']
X = df.drop(columns=['medal'])

# Split the data into training and testing sets
# CHANGED RANDOM STATE TO HAVE A LARGER NUMBER OF VALUES IN TEST DF
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=19)

# Initialize classifiers
classifiers = {
    "Random Forest": RandomForestClassifier(),
    "AdaBoost": AdaBoostClassifier(),
    "SVM": SVC(),
    "K-Nearest Neighbors": KNeighborsClassifier(),
    "Decision Tree": DecisionTreeClassifier(),
    "Naive Bayes": GaussianNB(),
    "Neural Network": MLPClassifier(),
}

results = {}

# Iterate through each classifier
for name, clf in classifiers.items():
    # Create a pipeline with StandardScaler for classifiers that require it
    if name in ["SVM", "K-Nearest Neighbors", "Neural Network"]:
        clf = make_pipeline(StandardScaler(), clf)

    # Fit the model to the training data
    clf.fit(X_train, y_train)

    # Make predictions on the test data
    y_pred = clf.predict(X_test)

    # Evaluate the model's performance
    accuracy = accuracy_score(y_test, y_pred)
    report = classification_report(y_test, y_pred)
    conf_matrix = confusion_matrix(y_test, y_pred)

    results[name] = {
        "accuracy": accuracy,
        "classification_report": report,
        "confusion_matrix": conf_matrix,
    }

# Print the results
for name, result in results.items():
    print(f"Classifier: {name}")
    print(f"Accuracy: {result['accuracy']:.4f}")
    print(f"Classification Report:\n{result['classification_report']}")
    print(f"Confusion Matrix:\n{result['confusion_matrix']}\n")


/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1469: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1469: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1469: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, m

Classifier: Random Forest
Accuracy: 0.9865
Classification Report:
              precision    recall  f1-score   support

           0       0.99      1.00      0.99        71
           1       1.00      0.67      0.80         3

    accuracy                           0.99        74
   macro avg       0.99      0.83      0.90        74
weighted avg       0.99      0.99      0.99        74

Confusion Matrix:
[[71  0]
 [ 1  2]]

Classifier: AdaBoost
Accuracy: 0.9865
Classification Report:
              precision    recall  f1-score   support

           0       0.99      1.00      0.99        71
           1       1.00      0.67      0.80         3

    accuracy                           0.99        74
   macro avg       0.99      0.83      0.90        74
weighted avg       0.99      0.99      0.99        74

Confusion Matrix:
[[71  0]
 [ 1  2]]

Classifier: SVM
Accuracy: 0.9595
Classification Report:
              precision    recall  f1-score   support

           0       0.96      1.0

/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  warnings.warn(
/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1469: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1469: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-

Classifier: Random Forest
Accuracy: 0.9865

Classifier: AdaBoost
Accuracy: 0.9865

Classifier: Decision Tree
Accuracy: 0.9865

In [7]:
X_train.head()

Rank    D      E   ND   Total  year  Name  Nation  round_final  \
65      6  5.9  8.500  0.0  14.400  2019   101      58        False   
54      3  5.7  8.750  0.0  14.450  2019     8      66         True   
284   167  4.0  7.966  0.0  11.966  2019    70      22        False   
248   131  4.8  8.033  0.0  12.833  2019    33      29        False   
263   146  5.0  7.500  0.0  12.500  2019    19      55        False   

     round_qual  
65         True  
54        False  
284        True  
248        True  
263        True

In [8]:
olymp.head()


Rank  D Score  E Score  Pen.   Total  year  Name  nation  round_AAfinal  \
0  54.0      4.4    8.066   0.0  12.466  2023   157      56          False   
1  12.0      5.8    8.400   0.0  14.200  2022   356      26           True   
2  26.0      5.8    8.500   0.0  14.300  2022   356      26          False   
3   9.0      5.8    8.600   0.0  14.400  2022   356      26          False   
4  10.0      5.8    8.266   0.0  14.066  2022   356      26          False   

   round_TeamFinal  round_final  round_qual  
0            False        False        True  
1            False        False       False  
2            False        False        True  
3            False        False        True  
4             True        False       False

In [9]:
olymp.rename(columns={'D Score': 'D', 'E Score': 'E',
                     'Pen.': 'ND', 'nation': 'Nation'}, inplace=True)

In [10]:
olymp=olymp.drop(columns=['round_TeamFinal', 'round_AAfinal'])

In [11]:
from sklearn.ensemble import AdaBoostClassifier
from sklearn.ensemble import RandomForestClassifier

base_estimator = RandomForestClassifier(n_estimators=100, max_depth=3)
ada_clf = AdaBoostClassifier(base_estimator=base_estimator, n_estimators=1000)


In [12]:

# Assuming you already have your DataFrame 'df' and the target variable is 'medal'
y = df['medal']
X = df.drop(columns=['medal'])

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=19)

# Create a custom base estimator (e.g., RandomForestClassifier)
base_estimator = RandomForestClassifier(n_estimators=10, max_depth=30)

# Initialize the AdaBoost classifier with the custom base estimator
clf = AdaBoostClassifier(base_estimator=base_estimator, n_estimators=1000)

# Fit the model to the training data
clf.fit(X_train, y_train)

# Make predictions on the test data
y_pred = clf.predict(X_test)

# Evaluate the model's performance
accuracy = accuracy_score(y_test, y_pred)
report = classification_report(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)

# Print the results
print("AdaBoost Classifier with a Custom Base Estimator (RandomForest)")
print(f"Accuracy: {accuracy:.4f}")
print(f"Classification Report:\n{report}")
print(f"Confusion Matrix:\n{conf_matrix}")

AdaBoost Classifier with a Custom Base Estimator (RandomForest)
Accuracy: 0.9865
Classification Report:
              precision    recall  f1-score   support

           0       0.99      1.00      0.99        71
           1       1.00      0.67      0.80         3

    accuracy                           0.99        74
   macro avg       0.99      0.83      0.90        74
weighted avg       0.99      0.99      0.99        74

Confusion Matrix:
[[71  0]
 [ 1  2]]


/Users/ryantalbot/opt/anaconda3/envs/tf2/lib/python3.9/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(


In [13]:
y_pred = clf.predict(olymp)


In [14]:
ypreds  = pd.Series(y_pred)
olymp['ypred']=ypreds


In [15]:
olymp['ypred'].value_counts()

ypred
0    1336
1       3
Name: count, dtype: int64

In [16]:
filtered_df = olymp[olymp['ypred'] == 1]

In [17]:
filtered_df['Name'].unique()

array([145,  94,  74])

In [23]:
name_encoded_values = [145,  94,  74]

# Filter the DataFrame to get the corresponding "Name" values
result = pbnames.loc[pbnames['Name_encoded'].isin(name_encoded_values), 'Name']


In [24]:
result.unique()

array(['ferhat arican', 'curran phillips', 'carlos edriel yulo'],
      dtype=object)

In [20]:
pbolymp["Name"]=pbolymp["Name"].str.lower()

In [25]:
names_to_filter = ['ferhat arican', 'curran phillips', 'carlos edriel yulo']

# Filter the DataFrame to get rows with the specified "Name" values
result = pbolymp[pbolymp['Name'].isin(names_to_filter)]


In [26]:
result['Country'].value_counts()

Country
TUR    41
PHI    40
USA    21
Name: count, dtype: int64

In [28]:
result[result['Country']=='USA']

Name Country        Date Gender  \
6153  curran phillips     USA  2023-03-12      m   
6154  curran phillips     USA  2023-03-12      m   
6155  curran phillips     USA  2023-03-12      m   
6156  curran phillips     USA  2023-03-12      m   
6352  curran phillips     USA  2022-08-20      m   
6353  curran phillips     USA  2022-08-20      m   
6354  curran phillips     USA  2022-08-20      m   
6355  curran phillips     USA  2022-08-20      m   
6356  curran phillips     USA  2022-08-20      m   
6357  curran phillips     USA  2022-08-20      m   
6358  curran phillips     USA  2022-02-27      m   
6359  curran phillips     USA  2022-02-27      m   
6360  curran phillips     USA  2022-02-27      m   
6361  curran phillips     USA  2023-08-27      m   
6362  curran phillips     USA  2023-08-27      m   
6363  curran phillips     USA  2023-08-27      m   
6364  curran phillips     USA  2023-08-27      m   
6365  curran phillips     USA  2023-08-27      m   
6366  curran phillips     USA  2023-08-27      m   
6367  curran phillips     USA  2023-02-26      m   
6368  curran phillips     USA  2023-02-26      m   

                       Competition    Round              Location Apparatus  \
6153  FIG Apparatus World Cup 2023    final      Baku, Azerbaijan        HB   
6154  FIG Apparatus World Cup 2023     qual      Baku, Azerbaijan        HB   
6155  FIG Apparatus World Cup 2023    final      Baku, Azerbaijan        PB   
6156  FIG Apparatus World Cup 2023     qual      Baku, Azerbaijan        PB   
6352       2022 U.S. Championships     qual             Tampa, FL        HB   
6353       2022 U.S. Championships    final             Tampa, FL        HB   
6354       2022 U.S. Championships     qual             Tampa, FL        PB   
6355       2022 U.S. Championships    final             Tampa, FL        PB   
6356       2022 U.S. Championships     qual             Tampa, FL        VT   
6357       2022 U.S. Championships    final             Tampa, FL        VT   
6358               2022 Winter Cup  AAfinal         Frisco, Texas        HB   
6359               2022 Winter Cup  AAfinal         Frisco, Texas        PB   
6360               2022 Winter Cup  AAfinal         Frisco, Texas        VT   
6361       2023 U.S. Championships     qual          San Jose, CA        HB   
6362       2023 U.S. Championships    final          San Jose, CA        HB   
6363       2023 U.S. Championships     qual          San Jose, CA        PB   
6364       2023 U.S. Championships    final          San Jose, CA        PB   
6365       2023 U.S. Championships     qual          San Jose, CA        VT   
6366       2023 U.S. Championships    final          San Jose, CA        VT   
6367               2023 Winter Cup  AAfinal  Louisville, Kentucky        HB   
6368               2023 Winter Cup  AAfinal  Louisville, Kentucky        PB   

      Rank  D_Score  E_Score  Penalty   Score  Total_Score  
6153   5.0      5.5    7.633      0.0  13.133       13.133  
6154   2.0      5.5    8.633      0.0  14.133       14.133  
6155   5.0      6.8    7.700      0.0  14.500       14.500  
6156   1.0      6.8    8.700      0.0  15.500       15.500  
6352   0.0      5.5    6.450      0.0  11.950       11.950  
6353   0.0      5.7    7.550      0.0  13.250       13.250  
6354   0.0      6.9    8.450      0.0  15.350       15.350  
6355   0.0      6.9    8.400      0.0  15.300       15.300  
6356   0.0      5.2    9.200      0.0  14.400       14.400  
6357   0.0      5.2    8.850      0.0  14.050       14.050  
6358   1.0      5.7    7.950      0.0  13.650       13.650  
6359   1.0      6.7    8.450      0.0  15.150       15.150  
6360   6.0      5.2    9.500      0.0  14.700       14.700  
6361   0.0      5.8    7.050      0.0  12.850       12.850  
6362   0.0      5.8    8.050      0.0  13.850       13.850  
6363   0.0      6.8    8.750      0.0  15.550       15.550  
6364   0.0      6.4    8.100      0.0  14.500       14.500  
6365   0.0      5.2    9.450      0